src/VTK/VtkIjkGridRepresentation.cxx

```diff
@@ -184,19 +184,99 @@ void VtkIjkGridRepresentation::addProperty(const std::string & uuidProperty, vtk
 }
 
 //----------------------------------------------------------------------------
-long VtkIjkGridRepresentation::getAttachmentPropertyCount(const std::string & uuid, const FesppAttachmentProperty propertyUnit)
+long VtkIjkGridRepresentation::getAttachmentPropertyCount(const std::string & uuid, const VtkEpcCommon::FesppAttachmentProperty propertyUnit)
 {
 	long result = 0;
-	resqml2_0_1::AbstractIjkGridRepresentation* ijkGridRepresentation = nullptr;
-	common::AbstractObject* obj = epcPackageRepresentation->getResqmlAbstractObjectByUuid(getUuid());
-	if (obj != nullptr && (obj->getXmlTag() == "IjkGridRepresentation" || obj->getXmlTag() == "TruncatedIjkGridRepresentation")) {
-		ijkGridRepresentation = static_cast<resqml2_0_1::AbstractIjkGridRepresentation*>(obj);
-		result = ijkGridRepresentation->getCellCount();
+	if (propertyUnit == VtkEpcCommon::CELLS)
+	{
+		common::AbstractObject* obj;
+		if (subRepresentation)
+		{
+			// SubRep
+			obj = epcPackageSubRepresentation->getResqmlAbstractObjectByUuid(getUuid());
+			auto subRepresentation1 = static_cast<resqml2::SubRepresentation*>(obj);
+			if (isHyperslabed)
+				result = iCellCount*jCellCount*(maxKIndex - initKIndex);
+			else
+				result = subRepresentation1->getElementCountOfPatch(0);
+		}else
+		{
+			// Ijk Grid
+			obj = epcPackageRepresentation->getResqmlAbstractObjectByUuid(getUuid());
+			auto ijkGridRepresentation = static_cast<resqml2_0_1::AbstractIjkGridRepresentation*>(obj);
+			result = ijkGridRepresentation->getCellCount();
+		}
 	}
 	return result;
 
 }
 
+//----------------------------------------------------------------------------
+int VtkIjkGridRepresentation::getICellCount(const std::string & uuid) const
+{
+	long result = 0;

```

Comment:
Same as above.
I would not instantiate result but directly return the result in the if, else and last line.